docs/docs/specs/v0/composer.md

```diff
@@ -0,0 +1,190 @@
+# Composer
+
+A developer-first, peer-to-peer knowledge management system.
+
+## Scenarios
+
+1. Users can collaborate on a github issue together
+2. Users can work on folders of markdown files
+3. Developers can extend Composer with custom data sources and visual surfaces
+
+## Extensibility
+
+> "Composer Chrome" refers to the total composer UI organization and all affordances surrounding the content the users interact with - all UI which is not the content itself.
+
+Two levels of complexity are proposed in sequence:
+
+**Static chrome**, where we constrain the application chrome to be statically defined and not extensible by plugins. Elements of the chrome may each expose some API surface for plugins to extend, but the overall elements of the chrome are fixed and can be assumed by all plugins without explicit dependency expressions.
+
+**Dynamic chrome**, where the chrome and it's organization is also extensible and now plugins have to express explicitly the elements of chrome they depend on because their presence can no longer be assumed.
+
+These are in addition to another orthogonal dimension of complexity, the way we load plugins:
+
+**Static plugins**, where a `<Composer />` element is given a list of `Plugin[]` directly as props and has to be compiled into the app at build time.
+
+**Dynamic plugins**, where the app knows how to load plugins via ESM `import` calls in the browser at runtime.
+
+### 1. Static chrome (v1)
+
+Large screens are split vertically into `sidebar` and `content` areas.
+
+```
+.------------.
+|    |     o |
+|    |       |
+|    |       |
+|    |       |
+'------------'
+```
+
+Small screens turn the `sidebar` into a slide-over element that hides by default.
+
+The HALO button (o) is in one of the corners and can be used to access the DXOS Shell.
+
+The sidebar presents a `tree` of nodes which can be populated by plugins.
+
+The `tree` is a logical first solution to the problem of organization and represents a model users and developers are very familiar with and have grown to expect, having found the same model in almost all nearby and competitive products: Notion, Quip, Dropbox Paper, Google Docs, VSCode, Obsidian, ... etc.
```

Comment:
I agree as long as the framework doesn't force this assumption. I.e., I'd like to flesh out a higherlevel application ontology (partly explained in this PR already) that discusses aspects of organization, navigation (incl. URL routes, breadcrumbs, links), search, suggest, primary content, and content aggregation (frames/stacks).

E.g., for composer: top-level data organization is managed by a (n-level?) tree, where the root nodes are X, and leaves are Y (see other note asking questions about messages, tasks, etc. vs tables, task lists). In this case is composer an example of an app that happens to put the Tree in a sidebar. Or is the sidebar/tree a core concept of the framework. Again, trying to tease out application vs framework.

Reply:
good discussion to have here.

I'm trying to make the position that the framework should not do certain things too abstractly too early, and should embrace a step function rewrite between v1 and v2. In v1 it's therefore correct to statically prescribe `tree` and `content`, and the big value step in v2 will be the unlocking of this to be more abstract. Lots of mileage is possible without the whole 9 yards here.

Added some more description about where content items go https://github.com/dxos/dxos/pull/3067/files#diff-c2e7a79caf635172b918b576878c6e1099e7f2ce5d02f89257ae4b504a799e13R168